### PR TITLE
feat: select mode resets on appstart + general `sessionState` sanitize

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves [MOPLAT-563] and [MOPLAT-312]

### Description

I took the `sanitize` func from eigen, and used it in ezpz to remove these from persisting.
I also moved the select mode state into that sessionstate, since this doesn't need persisting.

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-563]: https://artsyproduct.atlassian.net/browse/MOPLAT-563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MOPLAT-312]: https://artsyproduct.atlassian.net/browse/MOPLAT-312?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/system/store/Models/SelectModeModel.ts [hunk 1]
```diff
@@ -17,26 +18,37 @@ export interface SelectModeModel {
...
-      item: d,
-    })),
-  ]),
+  sessionState: {
+    isActive: false,
+    artworks: [],
+    installs: [],
+    documents: [],
+    items: computed((state) => {
+      console.log("COMPUTEDDDD", state)
```
Comment (rajsam003) [inline src/app/system/store/Models/SelectModeModel.ts:32]: 👀

**Review by rajsam003 — APPROVED** [review]
Looks good. Just need to remove the console.log :)

File: src/app/system/store/Models/SelectModeModel.ts [hunk 2]
```diff
@@ -17,26 +18,37 @@ export interface SelectModeModel {
   setSelectedItems: Action<this, { type: ItemType; items: Array<string> }>
 
   // supporting actions
-  setSelectMode: Action<this, this["isActive"]>
+  setSelectMode: Action<this, this["sessionState"]["isActive"]>
```
Comment (damassi) [inline src/app/system/store/Models/SelectModeModel.ts:21]: Loving this `this` discovery of yrs

File: src/app/system/store/Models/SelectModeModel.ts [hunk 3]
```diff
@@ -17,26 +18,37 @@ export interface SelectModeModel {
...
+    isActive: false,
+    artworks: [],
+    installs: [],
+    documents: [],
+    items: computed((state) => {
+      console.log("COMPUTEDDDD", state)
+      return [
+        ...state.artworks.map((a): { type: ItemType; item: string } => ({
+          type: "artwork",
+          item: a,
```
Comment (damassi) [inline src/app/system/store/Models/SelectModeModel.ts:36]: This is minor and non-blocking but would always encourage full variable names vs single letters -- if `a` is an artwork seeing 
```
state.artworks.map((artwork) => ...
```
only serves readability in the long-term

**Review by damassi — APPROVED** [review]

File: src/app/system/store/Models/SelectModeModel.ts (hunk 1 above)
Comment (pvinis) [reply src/app/system/store/Models/SelectModeModel.ts:32]: lol

**Pull request merged**